crates/api/src/fetch.rs

```diff
@@ -142,11 +141,8 @@ pub type DynFetchFactory = Arc<dyn FetchFactory>;
 /// Summary of the fetch state.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct FetchStateSummary {
-    /// The op ids that are currently being fetched.
-    ///
-    /// Each op id is associated with one or more peer URL from which the op data could be
-    /// requested.
-    pub pending_requests: HashMap<OpId, Vec<Url>>,
+    /// The number of op ids that are currently being fetched.
+    pub pending_request_count: usize,
```

Comment:
Which op ids are being fetched is no longer kept in state. I propose to only return the count of op ids that are being fetched. Otherwise I'd bring back the requests to the state.

Reply:
I wonder if "pending request [...]" is still the correct name for this field and whether we should think about the value that's left in it.
If I understand the changes of this PR correctly, then this count is now only tracking what's currently in the outgoing request tokio channel but not requested from other peers yet. As soon as a request is being sent to another peer, the count is decreased so requests that are pending in the sense that we're still expecting a response in the incoming response channel are not represented by this number.

Reply:
> how many ops are discovered but not there yet

That this is interesting I agree but that's not what I see the current PR actually doing if it reduces the counter immediately after sending a request.

Reply:
Yes, you're right, now it tracks something different. It needs to be tracked to notify listeners when the fetch request queue is drained.